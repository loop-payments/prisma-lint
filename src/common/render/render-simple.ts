import chalk from 'chalk';

import type { Violation } from '#src/common/violation.js';
import { keyViolationListPairs } from '#src/common/render/render-util.js';

export const renderViolationsSimple = (violations: Violation[]) => {
  const pairs = keyViolationListPairs(violations);
  return pairs
    .flatMap(([key, violations]) => {
      const first = violations[0];
      const location = first.field?.location ?? first.model.location;
      if (!location) {
        throw new Error('No location');
      }
      const { startLine, startColumn } = location;
      return [
        `  ${key} ${chalk.gray(`${startLine}:${startColumn}`)}`,
        ...violations.flatMap(renderViolationSimple),
      ];
    })
    .join('\n');
};

const renderViolationSimple = ({ ruleName, message }: Violation) => [
  `    ${chalk.red('error')} ${message} ${chalk.gray(`${ruleName}`)}`,
];
