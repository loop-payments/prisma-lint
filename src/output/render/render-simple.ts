import chalk from 'chalk';

import type { Violation } from '#src/common/violation.js';
import { keyViolationListPairs } from '#src/output/render/render-util.js';

export const renderViolationsSimple = (violations: Violation[]): string[] => {
  const pairs = keyViolationListPairs(violations);
  return pairs.flatMap(([key, violations]) => {
    const first = violations[0];
    const location =
      first.field?.location ?? first.model?.location ?? first.enum?.location;
    if (!location) {
      throw new Error('No location');
    }
    const { startLine, startColumn } = location;
    return [
      `  ${key} ${chalk.gray(`${startLine}:${startColumn}`)}`,
      ...violations.flatMap(renderViolationSimple),
    ];
  });
};

const renderViolationSimple = ({ ruleName, message }: Violation) => [
  `    ${chalk.red('error')} ${message} ${chalk.gray(`${ruleName}`)}`,
];
