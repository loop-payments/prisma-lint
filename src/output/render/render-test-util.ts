import type { Model } from '@mrleebo/prisma-ast';

import type { Violation } from '#src/common/violation.js';

export const MOCK_SOURCE_CODE = `model Foo {
  id     Int    @id
  displayName String
}`;

const model: Model = {
  type: 'model',
  name: 'Foo',
  properties: [],
  location: {
    startLine: 1,
    startColumn: 1,
    startOffset: 1,
    endLine: 1,
    endColumn: 9,
    endOffset: 1,
  },
};
const fileName = 'test.prisma';
export const MOCK_VIOLATIONS: Violation[] = [
  {
    ruleName: 'fake-model-rule-name',
    message: 'Fake model rule violation message',
    fileName,
    model,
  },
  {
    ruleName: 'fake-field-rule-name',
    message: 'Fake field rule violation message',
    fileName,
    model,
    field: {
      type: 'field',
      name: 'displayName',
      fieldType: 'String',
      location: {
        startLine: 2,
        startColumn: 3,
        startOffset: 1,
        endLine: 2,
        endColumn: 12,
        endOffset: 1,
      },
    },
  },
];
