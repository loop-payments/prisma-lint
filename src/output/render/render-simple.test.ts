import { renderViolationsSimple } from '#src/output/render/render-simple.js';
import { MOCK_VIOLATIONS } from '#src/output/render/render-test-util.js';

describe('render json', () => {
  it('matches snapshot', () => {
    const result = renderViolationsSimple(MOCK_VIOLATIONS);
    expect(result).toBeDefined();
    expect(result).toMatchSnapshot();
  });
});
