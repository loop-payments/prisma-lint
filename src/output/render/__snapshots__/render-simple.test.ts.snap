// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render json matches snapshot 1`] = `
[
  "  Foo [90m1:1[39m",
  "    [31merror[39m Fake model rule violation message [90mfake-model-rule-name[39m",
  "  Foo.displayName [90m2:3[39m",
  "    [31merror[39m Fake field rule violation message [90mfake-field-rule-name[39m",
]
`;
